File: src/test/java/com/polarbookshop/edgeservice/EdgeServiceApplicationTests.java
package com.polarbookshop.edgeservice;

import org.junit.jupiter.api.Test;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.context.annotation.Import;
import org.testcontainers.junit.jupiter.Testcontainers;

@Import(TestcontainersConfiguration.class)
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
@Testcontainers
class EdgeServiceApplicationTests {

  @Test
  void contextLoads() {}
}

File: src/test/kotlin/com/polarbookshop/edgeservice/EdgeServiceApplicationTests.kt
package com.polarbookshop.edgeservice

import com.ninjasquad.springmockk.MockkBean
import org.junit.jupiter.api.Test
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.context.annotation.Import
import org.springframework.security.oauth2.client.registration.ReactiveClientRegistrationRepository

@Import(TestcontainersConfiguration::class)
@SpringBootTest
class EdgeServiceApplicationTests {
    @MockkBean lateinit var clientRegistrationRepository: ReactiveClientRegistrationRepository

    @Test
    fun contextLoads() {
    }
}
